virttest/qemu_devices/qdevices.py

```diff
@@ -1105,6 +1116,35 @@ def verify_hotplug(self, out, monitor):
         else:
             return False
 
+    def _cmdline_json(self):
+        command_dict = {}
+        out = "-%s " % self.type
+        for key, val in six.iteritems(self.params):
+            # wwn needs to be presented as hexadecimal
+            if key in ["wwn"]:
+                command_dict[key] = int(val, 16)
+            # physical_block_size from device ("driver": "scsi-hd")
+            # logical_block_size from device ("driver": "scsi-hd")
+            # bootindex from device ("driver": "scsi-hd")
+            # max_sectors from device ("driver": "virtio-scsi-pci")
+            # num_queues from device ("driver": "virtio-scsi-pci")
+            # virtqueue_size from device ("driver": "virtio-scsi-pci")
+            elif key in ["physical_block_size", "logical_block_size",
+                         "bootindex", "max_sectors", "num_queues",
+                         "virtqueue_size"]:
+                command_dict[key] = int(val.split(" ")[0])
+            # port from device ("driver": "usb-storage")
+            elif key == "port":
+                command_dict[key] = str(val)
```

Comment:
Do we want to add a check of usb-storage here? in case there is some other 'port' params defined, and a int number is required.

Reply:
> Do we want to add a check of usb-storage here? in case there is some other 'port' params defined, and a int number is required.

So far, it's NOT necessary.
Based on the devices supported, those codes work well, which are confirmed by owner.